File: Solutions/_9.cs
public class Solution {
    public bool IsPalindrome(int x) {
        
        if (x < 0) return false;
        if (x == 0) return true;
        
        var intergerChar = x.ToString().ToCharArray();
        var pointerRight = intergerChar.Length - 1;
        var pointerLeft = 0;
        
        while(pointerLeft < pointerRight)
        {
            if(intergerChar[pointerLeft] != intergerChar[pointerRight] )
            {
                return false;
            }
            
            pointerRight--;
            pointerLeft++;
        }
        
        return true; 
    }
}

File: Solutions/_9.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    if (x < 0 ) return false; 
    if(x == 0) return true;

    let num = x;
    let pointerLeft = 0;
    let pointerRight = x.toString().length - 1;

    while(pointerLeft < pointerRight){
        if(num.toString()[pointerLeft] != num.toString()[pointerRight]) return false;
        pointerLeft++;
        pointerRight--;
    } 
    return true; 
};